### ISSUE TITLE
catala-language-server: test case format should include prose sections and possibly metadata

### ISSUE THREAD

**Issue by rprimet** [CatalaLang/catala#673]
The current format for test cases (those that are rendered by the UI aimed at non-programmers) is a list of `Test` instances, each of which is a function invocation (with inputs and a list of assertions on the return value). 

We should also include comments and possibly metadata on the test fields, as those are highly valued by test writers and readers (see the catala seminar on payfit for an example).

This issue is there to discuss what we want to support in terms of prose and/or metadata

**Comment by denismerigoux** [comment]
I agree on the need for metadata and comments in tests. We already have some examples of what a fully commented and qualified test looks like on the income tax study : see e.g. page 187 [impot_revenu.pdf](https://github.com/user-attachments/files/16848490/impot_revenu.pdf). Because Catala uses literate programming, I feel that the right place for explaining what a test case should do is within the Catala file itself, as Mardown text before the snippet of Catala code containing the code of the test case scope.

Concerning more localized comments on the test fields, they can be stored as comments in the Catala code snippets. This might be a good occasion to kickstart the normalization effort of Catala code comments with a proper attribute language that @rprimet's IDE could pick up (see https://github.com/CatalaLang/catala/issues/392).
